gdk4/src/subclass/content_provider.rs

```diff
@@ -0,0 +1,248 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::{Clipboard, ContentFormats, ContentProvider};
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::{Cast, Object, Value};
+
+pub trait ContentProviderImpl: ContentProviderImplExt + ObjectImpl {
+    fn content_changed(&self, provider: &Self::Type) {
+        self.parent_content_changed(provider)
+    }
+
+    fn attach_clipboard(&self, provider: &Self::Type, clipboard: &Clipboard) {
+        self.parent_attach_clipboard(provider, clipboard)
+    }
+
+    fn detach_clipboard(&self, provider: &Self::Type, clipboard: &Clipboard) {
+        self.parent_detach_clipboard(provider, clipboard)
+    }
+
+    fn ref_formats(&self, provider: &Self::Type) -> ContentFormats {
+        self.parent_ref_formats(provider)
+    }
+
+    fn ref_storable_formats(&self, provider: &Self::Type) -> ContentFormats {
+        self.parent_ref_storable_formats(provider)
+    }
+
+    // fn write_mime_type_async(&self, provider: &Self::Type) -> ContentFormats {
+    //     self.parent_write_mime_type_async(provider)
+    // }
+
+    fn value(&self, provider: &Self::Type, value: &mut Value) -> Result<(), glib::Error> {
```

Comment:
Do you have some example usage in C somewhere with the implementation of this and a caller of `gdk_content_provider_get_value`? This seems a bit strange and the documentation is also not very helpful

Reply:
See for ex: https://gitlab.gnome.org/GNOME/gtk/-/blob/master/gtk/gtklabel.c#L4545

The content provider checks the expected value type and fills it.

Reply:
And https://gitlab.gnome.org/GNOME/gtk/-/blob/master/gdk/gdkclipboard.c#L782

Reply:
Should this then maybe become `fn value(&self, provider, type_: glib::Type) -> Result<glib::Value, glib::Error>`? That would be nicer to use from a Rust point of view at least.

The C API seems to be just too lazy to separate the in parameter (type) and out parameter (value), so both are put into the same.

Reply:
Yes, that's probably more convenient.